spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/types/external.scala

```diff
@@ -79,6 +79,38 @@ object external {
     new File(path).getAbsoluteFile().toURI()
   }
 
+  // ================ HBase entities ======================
+  val HBASE_NAMESPACE_STRING = "hbase_namespace"
+  val HBASE_TABLE_STRING = "hbase_table"
+  val HBASE_COLUMNFAMILY_STRING = "hbase_column_family"
+  val HBASE_COLUMN_STRING = "hbase_column"
+
+  private def stripNameSpace(tableName: String): String = {
+    tableName.substring(tableName.indexOf(":") + 1)
+  }
+
+  def hbaseTbUniqueAttribute(cluster: String, tableName: String, nameSpace: String) =
+    nameSpace.toLowerCase + '.' + stripNameSpace(tableName.toLowerCase()) + '@' + cluster
+
+  def hbaseTableToEntity(cluster: String, tableName: String, nameSpace: String): Seq[AtlasEntity] = {
+    val hbaseEntity = new AtlasEntity(HBASE_TABLE_STRING)
+    hbaseEntity.setAttribute(AtlasClient.REFERENCEABLE_ATTRIBUTE_NAME,
+      hbaseTbUniqueAttribute(cluster, tableName, nameSpace))
+    hbaseEntity.setAttribute(AtlasClient.NAME, tableName.toLowerCase)
+    hbaseEntity.setAttribute(AtlasConstants.CLUSTER_NAME_ATTRIBUTE, cluster)
+    hbaseEntity.setAttribute("uri", nameSpace.toLowerCase + ":" + tableName.toLowerCase)
```

Comment:
Is there any rule about the case we use ":" (line 101) and the case we use "." (line 93).

Reply:
"uri" is optional.  I confirmed its format with Atlas team. Per the example they showed to me, they used ":" instead of ".".

Reply:
Can we reuse Atlas function here?

Reply:
It seems like "uri" does not have any rule. I make its format to "namespace:table" is just follow the example Atlas team showed to me. I searched some other examples, e.g. https://gist.github.com/coheigea/e32002058cf530eded145d6963c4c264, the format of "uri" is different.

Reply:
The best way is having a single function for the single purpose. If possible, upgrade Atlas first and reuse the given function.

Reply:
I did not find some function in Atlas-Hbase Bridge to get the "uri". I think the reason why it provides getUniqueAttribute() in Atlas-Hbase is that the unique attribute has to be consistent as it is used to do de-duplication.